src/plugins/validation/2and3/semantic-validators/walker-ibm.js

```diff
@@ -49,17 +49,6 @@ module.exports.validate = function({ jsSpec, resolvedSpec }, config) {
         }
       }
     }
-
-    // check for and flag null values - they are not allowed by the spec and are likely mistakes
-    Object.keys(obj).forEach(key => {
-      if (obj[key] === null) {
-        messages.addMessage(
-          [...path, key],
-          'Null values are not allowed for any property.',
-          'error'
-        );
-      }
-    });
```

Comment:
Rather than removing the check completely, I wonder if we should keep the check but allow `null` values on certain fields. From what I understand, null values are allowed on the `default` field or any annotations (fields that start with `x-`). 

Or, at the very least, just make it configurable so that users can turn it off if they need. We may want to make this a warning too, if our justification is they "are likely mistakes".

My suggestion might be to do all 3: allow certain fields, make it configurable, change the default to a warning. What do you think @mkistler ?

Reply:
I would also like to see a more surgical fix.  We can address #155 by simply changing the condition in line 55 to
```
      if (key !== "default" && obj[key] === null) {
```